impl/httpMetricsManager.go

```diff
@@ -2,24 +2,62 @@ package impl
 
 import (
 	"fmt"
+	"sort"
+	"strings"
 	"time"
 
+	"github.com/satori/go.uuid"
 	"github.com/xiaoyao1991/chukonu/core"
 )
 
 type HttpMetricsManager struct {
 	requestSent int
 	countQueue  chan int
+	records     HttpRecords
+	durationSum DurationSummation
+	numOfError  int
 	// throughputRecord []int
 }
 
+type HttpRecord struct {
+	duration time.Duration
+	id       uuid.UUID
+	status   string
+}
+type HttpRecords []HttpRecord
+
+type DurationSummation struct {
+	sum   time.Duration
+	index int
+}
+
+// ------------sorting interface for struct HttpRecord-----------
+func (slice HttpRecords) Len() int {
+	return len(slice)
+}
+func (slice HttpRecords) Less(i, j int) bool {
+	return slice[i].duration < slice[j].duration
+}
+func (slice HttpRecords) Swap(i, j int) {
+	slice[i], slice[j] = slice[j], slice[i]
+}
+
+//----------------------------------------------------------------
+
 func (m *HttpMetricsManager) RecordRequest(request core.ChukonuRequest) {
 }
 func (m *HttpMetricsManager) RecordResponse(response core.ChukonuResponse) {
+	record := new(HttpRecord)
```

Comment:
I don't see `new` being used a lot. Normally I'd just `var record HttpRecord`, or `record := HttpRecord{}`, and an empty struct will be created, or have a `NewHttpRecord` method that act like a contructor